nuget.config

```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<configuration>
+
+  <packageSources>
+    <add key="nuget" value="https://api.nuget.org/v3/index.json" />
+    <add key="github-open-feature" value="https://nuget.pkg.github.com/open-feature/index.json" />
+  </packageSources>
+
+  <packageSourceMapping>
+    <packageSource key="nuget">
+      <package pattern="OpenFeature" />
+      <package pattern="OpenFeature.*" />
+      <package pattern="*" />
+    </packageSource>
+    <packageSource key="github-open-feature">
+      <package pattern="OpenFeature" />
+      <package pattern="OpenFeature.*" />
+    </packageSource>
+  </packageSourceMapping>
+
+</configuration>
```

Comment:
> [!NOTE]
>
> Following open-feature/dotnet-sdk#173, it is now possible for this repo to consume pre-release packages from `open-feature/dotnet-sdk` 🎉 
> 
> This file provides `dotnet restore` with a mapping of where to look for packages. For example, if a project in this repo references a pre-release version of `OpenFeature`, `dotnet restore` will first look in the `nuget` source, and when the pre-release version is not found, it will proceed to checking in the `github-open-feature` source.
> 
> Currently, this repo ___does not yet___ reference any pre-release package versions, so there is no action required ___today___ for contributors.
> 
> However, contributors can prepare their local environments using the following instructions:
> 
> 
> 1. Acquire a [GitHub personal access token (PAT)](https://docs.github.com/github/authenticating-to-github/creating-a-personal-access-token) scoped for `read:packages` and verify the permissions:
>    ```console
>    $ gh auth login --scopes read:packages
> 
>    ? What account do you want to log into? GitHub.com
>    ? What is your preferred protocol for Git operations? HTTPS
>    ? How would you like to authenticate GitHub CLI? Login with a web browser
> 
>    ! First copy your one-time code: ****-****
>    Press Enter to open github.com in your browser...
> 
>    ✓ Authentication complete.
>    - gh config set -h github.com git_protocol https
>    ✓ Configured git protocol
>    ✓ Logged in as ********
>    ```
> 
>    ```console
>    $ gh auth status
> 
>    github.com
>      ✓ Logged in to github.com as ******** (~/.config/gh/hosts.yml)
>      ✓ Git operations for github.com configured to use https protocol.
>      ✓ Token: gho_************************************
>      ✓ Token scopes: gist, read:org, read:packages, repo, workflow
>    ```
> 2. Run the following command to configure your local environment to consume packages from GitHub Packages:
>    ```console
>    $ dotnet nuget update source github-open-feature --username $(gh api user --jq .email) --password $(gh auth token) --store-password-in-clear-text
> 
>    Package source "github-open-feature" was successfully updated.
>    ```
> 
> 
> ## Related
> 
> - https://cli.github.com
> - https://learn.microsoft.com/nuget/consume-packages/package-source-mapping